src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -742,6 +742,13 @@ void RegisterAppInterfaceRequest::Run() {
   SetupAppDeviceInfo(application);
   auto status_notifier = AddApplicationDataToPolicy(application);
 
+  auto on_app_registered = [application](plugin_manager::RPCPlugin& plugin) {
+    plugin.OnApplicationEvent(plugin_manager::kApplicationRegistered,
+                              application);
+  };
+  application_manager_.ApplyFunctorForEachPlugin(on_app_registered);
```

Comment:
In this case it is important to remember that when the plugin received `kApplicationRegistered` event, the application is not finally registered. I would add this as a comment in the `kApplicationRegistered` enum

Reply:
Noting that this placement is the most important part of the change. This event needed to happen before an app was added to the applications accessor in the app manager.